src/riak_kv_qry_compiler.erl

```diff
@@ -97,16 +97,22 @@ run_select(Select, Row, InitialState) ->
 
 %% @priv
 run_select2([], _, _, Acc) ->
```

Comment:
@javajolt no, it has the same complexity, but running `lists:reverse` once while accumulating and at the end is slower.

If we used something else than [] for nulls then we could just do a
````erlang
[ Fn(Row, RowState] | run_select2(SelectTail, Row, RowState) ]
````
in `run_select2`, but then we'd have to flatten the list at the end. I think that is as expensive as `lists:reverse` and `++`.